client/components/add-commuters.js

```diff
@@ -0,0 +1,163 @@
+    group: PropTypes.object,
+    organizationId: PropTypes.string
+  }
+
+  componentWillMount () {
+    if (this.props.appendMode) {
+      this.setState({
+        name: this.props.group.name,
+        existingCommuters: this.props.existingCommuters
+      })
+    } else {
+      this.state = {
+        organizationId: this.props.organizationId
+      }
+    }
+  }
+
+  handleChange = (name, event) => {
+    this.setState({ [name]: event.target.value })
+  }
+
+  handleSubmit = () => {
+    const {appendMode, createCommuter, createGroup} = this.props
+    const commutersToCreate = this.state.newCommuters
+      ? this.state.newCommuters.map((commuter) => omit(commuter, '_id'))
+      : []
+
+    if (appendMode) {
+      createCommuter(commutersToCreate)
+    } else {
+      const newGroup = {...this.state}
+      newGroup.commuters = commutersToCreate
+      delete newGroup.newCommuters
+      createGroup(newGroup)
+    }
+  }
+
+  makeCommuterTable (commuters) {
+    return (
+      <BootstrapTable data={commuters}>
+        <TableHeaderColumn dataField='_id' isKey hidden />
+        <TableHeaderColumn dataField='name'>Name</TableHeaderColumn>
+        <TableHeaderColumn dataField='email'>Email</TableHeaderColumn>
+        <TableHeaderColumn dataField='address'>Address</TableHeaderColumn>
+      </BootstrapTable>
+    )
+  }
+
+  onDrop = (files) => {
+    const {appendMode, group} = this.props
+    const r = new FileReader()
+
+    r.onload = (e) => {
+      const newCommuters = csvParse(e.target.result, (row) => {
+        const {address, email, name} = row
+        const _id = row._id || uuid.v4()
+        // TODO: parse more field possibilities (first name, last name, etc)
+        const newCommuter = {address, email, _id, name}
+        if (appendMode) {
+          newCommuter.groupId = group._id
+        }
+        return newCommuter
+      })
+
+      this.setState({newCommuters})
+    }
+
+    // TODO: handle multiple files
+    r.readAsText(files[0])
+  }
+
+  render () {
+    const {appendMode, group, organizationId} = this.props
+    const groupName = this.state.name
+    const showAccordion = !!(this.state.existingCommuters || this.state.newCommuters)
+    return (
+      <Grid>
+        <Row>
+          <Col xs={12}>
+            <h3>
+              <span>{appendMode ? 'Add Commuters to Group' : 'Create Commuter Group'}</span>
+              <Button className='pull-right'>
+                <Link to={appendMode ? `/group/${group._id}` : `/organization/${organizationId}`}>
+                  <Icon type='arrow-left' />
+                  <span>Back</span>
+                </Link>
+              </Button>
+            </h3>
+            <form>
+              <FieldGroup
+                label='Name'
+                name='name'
+                onChange={this.handleChange}
+                placeholder='Enter name'
+                type='text'
+                value={groupName}
+                />
+              <Dropzone
```

Comment:
It's adding complexity without adding a feature. Those should be avoided really as much as possible with this project.